### PR TITLE
Fix "deprecated" function godoc comments to match standard formatting

### PR DESCRIPTION
**Pull request by Chyroc**

### CODE REVIEW

**Comment by thelinuxfoundation** [comment]
Thank you for your pull request. Before we can look at your contribution, we need to ensure all contributors are covered by a Contributor License Agreement.

After the following items are addressed, please respond with a new comment here, and the automated system will re-verify.

* User @Chyroc isn't covered by a CLA. They will need to complete the form at https://identity.linuxfoundation.org/projects/cncf

_Regards,
CLA GitHub bot_

**Comment by Chyroc** [comment]
@thelinuxfoundation done.

File: metadata/metadata.go [hunk 1]
```diff
@@ -28,7 +28,8 @@ import (
 	"golang.org/x/net/context"
 )
 
-// DecodeKeyValue returns k, v, nil.  It is deprecated and should not be used.
+// DecodeKeyValue returns k, v, nil.
+// Deprecated; should not be used.
```
Comment (dfawley) [inline metadata/metadata.go]: Please use the "standard" deprecated syntax, so static analysis tools will notice it correctly:

```
// DecodeKeyValue returns k, v, nil.
//
// Deprecated: use k and v directly instead.
```

**Comment by dfawley** [comment]
I have a feeling Travis won't like this, since our tests still call many "deprecated" functions.  You might want to limit it to just the one function for now.

**Comment by Chyroc** [comment]
@dfawley 

ci falied because of `benchmark/grpc_testing/services.pb.go:49:9: grpc.Invoke is deprecated: Use ClientConn.Invoke instead.  (SA1019)` (https://travis-ci.org/grpc/grpc-go/jobs/371845620#L540)

but it is generated by https://github.com/golang/protobuf/blob/master/protoc-gen-go/grpc/grpc.go#L290

should i fix protoc-gen-go or add it to `staticcheck -ignore` ?

**Comment by dfawley** [comment]
@Chyroc 

Thanks for the extensive fixes!

> should i fix protoc-gen-go or add it to staticcheck -ignore ?

We should revert the deprecation tag on it until after protoc-gen-go is updated to not use it.  I wouldn't want to set off other people's staticchecks.

**Comment by Chyroc** [comment]
make a pr( https://github.com/golang/protobuf/pull/589 ) to update protoc-gen-go to not use it

**Pull request merged**